pkg/manifests/manifests.go

```diff
@@ -634,6 +634,42 @@ func (f *Factory) PrometheusK8sEtcdServiceMonitor() (*monv1.ServiceMonitor, erro
 	return s, nil
 }
 
+func (f *Factory) PrometheusK8sEtcdSecret(tlsClient *v1.Secret, ca *v1.ConfigMap) (*v1.Secret, error) {
+	data := make(map[string]string)
+
+	for k, v := range tlsClient.Data {
+		data[k] = string(v)
```

Comment:
instead of doing these conversions from []byte to string and then back to []byte, consider getting and setting using the `stringData` field of the Secret. This way, you don't even need to construct an intermediate `data` map.

Reply:
agreed, fixed

Reply:
oh wait, StringData can only be used for writing:
```
	// stringData allows specifying non-binary secret data in string form.
	// It is provided as a write-only convenience method.
```

Reply:
ok, at least changed the write-path

Reply:
ah too bad :/